.github/workflows/new-issue-labeler.yml

```diff
@@ -0,0 +1,20 @@
+name: New issue triage
+on:
+  issues:
+    types: [opened]
+
+jobs:
+  label:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Label new issues
+      uses: actions/github-script@v7
```

Comment:
There is an official action for this: https://github.com/github/issue-labeler

The workflow can be made optional by using the context variables.

Reply:
Thanks for sharing. Updated the PR!

While we can use context variables here, I'm not sure if there's a way to get the role of an issue author using that.